### PR TITLE
Add IPv6 support

### PR DESCRIPTION
**Pull request by alex-schneider**
Describe your changes!

---
<details>
    <summary>Reviewer checklist</summary>
    <ul>
        <li>Read PR description: a summary about the changes is required</li>
        <li>Changelog updated</li>
        <li>Documentation: docs/{Reference, Cli, ...}, Docker and cli help/usage</li>
        <li>Pulled branch, manually tested</li>
        <li>Verified requirements are met</li>
        <li>Reviewed the code</li>
        <li>Reviewed the related tests</li>
    </ul>
</details>

### CODE REVIEW

File: docs/website/content/2.configuration/4.block/settings.md [hunk 1]
```diff
@@ -12,6 +12,12 @@ values: [
     "name": "accept_forwarded_url",
     "type": "tuple (string)"
   },
+  {
```
Comment (johakoch) [inline docs/website/content/2.configuration/4.block/settings.md:15]: Add new settings to docs/website/content/2.configuration/0.command-line.md, too.

File: config/settings.go [hunk 2]
```diff
@@ -33,10 +33,13 @@ var defaultSettings = Settings{
 // Settings represents the <Settings> object.
 type Settings struct {
 	AcceptForwarded *AcceptForwarded
+	BindAddresses   map[string]string
 	Certificate     []byte
 
 	CAFile                    string `hcl:"ca_file,optional" docs:"Adds the given PEM encoded CA certificate to the existing system certificate pool for all outgoing connections."`
 	AcceptForwardedURL        List   `hcl:"accept_forwarded_url,optional" docs:"Which {X-Forwarded-*} request HTTP header fields should be accepted to change the [request variables](../variables#request) {url}, {origin}, {protocol}, {host}, {port}. Valid values: {\"proto\"}, {\"host\"} and {\"port\"}. The port in a {X-Forwarded-Port} header takes precedence over a port in {X-Forwarded-Host}. Affects relative URL values for [{sp_acs_url}](saml) attribute and {redirect_uri} attribute within [{beta_oauth2}](oauth2) and [{oidc}](oidc)."`
+	BackendIPv6               bool   `hcl:"backend_ipv6,optional" docs:"Enables IPv6 support for all backends."`
+	BindAddress               string `hcl:"bind_address,optional" docs:"A list of addresses to bind." default:"*"`
```
Comment (johakoch) [inline config/settings.go:42]: What sort of list in a string is this? comma-separated? space-separated?

File: server/http.go [hunk 3]
```diff
@@ -32,7 +32,7 @@ type muxers map[string]*Mux
 type HTTPServer struct {
 	commandCtx context.Context
 	evalCtx    *eval.Context
-	listener   net.Listener
+	listener   []net.Listener
```
[the remark this answers is not in the document]
Comment (alex-schneider) [reply server/http.go]: Done

File: config/settings.go (hunk 2 above)
Comment (alex-schneider) [reply config/settings.go]: Done

File: docs/website/content/2.configuration/4.block/settings.md (hunk 1 above)
Comment (alex-schneider) [reply docs/website/content/2.configuration/4.block/settings.md:15]: Done

**Review by malud — APPROVED** [review]
tested with -bind-address :: and logs showed `client_ip="::1"` with `localhost`. Also ipv4 in logs while calling ipv4 (127.0.0.1).

**Pull request merged**